src/engine/guidance/post_processing.cpp

```diff
@@ -897,17 +897,27 @@ void trimShortSegments(std::vector<RouteStep> &steps, LegGeometry &geometry)
         // remove the initial distance value
         geometry.segment_distances.erase(geometry.segment_distances.begin());
 
+        const auto offset = zero_length_step ? geometry.segment_offsets[1] : 1;
+
         // We have to adjust the first step both for its name and the bearings
         if (zero_length_step)
         {
+            // since we are not only checking for epsilon but for a full meter, we can have multiple
+            // coordinates here
+            for (int count = 1; count < offset; ++count)
+            {
+                geometry.locations.erase(geometry.locations.begin());
```

Comment:
Instead of the loop you can use the range version of `erase`:

`geometry.location.erase(geometry.locations.begin(), geometry.locations.begin() + offset);`